test/rules/no-promise-as-boolean/test.ts.lint

```diff
@@ -0,0 +1,54 @@
+[typescript]: >= 2.4.0
+
+async function promiseFunc() {
```

Comment:
We'll need some more test cases to really exercise this rule's edge cases. The following should be included:

* `Promise | undefined`
* `Promise | false`
* `Promise | null`

Additionally, the `Array.forEach` case in the linked issue is missing.

Reply:
The extra edge cases are added, but I won't add the `Array.forEach` one, since that's more specific about not passing Promises in normal callbacks. I can try to add another rule for this, since this can be a bug (but not always is). 

The `no-promise-as-boolean` rule is always a bug, unless someones typings are wrong. :-)